src/mpid/ch4/netmod/ucx/ucx_vci.c

```diff
@@ -0,0 +1,52 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2016 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Mellanox Technologies Ltd.
+ *  Copyright (C) Mellanox Technologies Ltd. 2016. ALL RIGHTS RESERVED
+ */
+#include <mpidimpl.h>
+#include "ucx_impl.h"
+#include "ucx_types.h"
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_UCX_vci_alloc
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+void MPIDI_UCX_vci_alloc(int resource, int type, int properties, int *vci)
+{
+    if (type & MPIDI_VCI_EXCLUSIVE)
+        *vci = -1;
```

Comment:
Instead of hardcoding `-1`, would it be better to use a constant definition, e.g, `#define <prefix>_VCI_INVALID (-1)`? It may be defined at CH4 level so that all NMs can use it.

Reply:
Yes, I agree. In general, this PR needs to be re-written based on reviews on other `per-vci` related PRs.